activerecord/test/cases/relations_test.rb

```diff
@@ -677,6 +677,44 @@ def test_loading_with_one_association_with_non_preload
     assert_equal Post.find(1).last_comment, post.last_comment
   end
 
+  def test_load_associations
+    posts = Post.all
+    even_posts, odd_posts = posts.partition.each_with_index{ |p, i| i.even? }
+
+    assert_queries(1) do
+      preloaded_posts = Post.load_associations(even_posts, :comments)
```

Comment:
Since `load_associations` mutates the posts would it be clearer if we just referenced `even_posts` throughout?

Reply:
I also find the argument order hard to understand and I had to read it several times.

Would it be clearer as `Post.preload_associations(:comments, on: even_posts)`? A `preload_association` alias might also be nice 😊